geom/alg_set_op.go

```diff
@@ -72,3 +87,15 @@ func setOp(a, b Geometry, include func([2]label) bool) (Geometry, error) {
 	}
 	return g, nil
 }
+
+func merge(gc GeometryCollection) (Geometry, error) {
+	merged := Geometry{}
+	for _, g := range gc.geoms {
```

Comment:
Just noting that it's valid for a child in a geometry collection to be itself a geometry collection. This nesting can be arbitrarily deep. So in some sense you can think of a geometry collection as a tree with the root being the original geometry collection and any non-leaf nodes below it being additional geometry collections.

The code you've got here should be ok under the assumption that `Union` works for geometry collections.